### ISSUE TITLE
Discrepancy with placeholders without associated context enty

### ISSUE THREAD

**Issue by Maks3w** [php-fig/log#33]
PSR-3 says

> Placeholder names MUST correspond to keys in the context array.

But `LoggerInterfaceTest::testContextReplacement()`

```php
$logger->info('{Message {nothing} {user} {foo.bar} a}', array('user' => 'Bob', 'foo.bar' => 'Bar'));
$expected = array('info {Message {nothing} Bob Bar a}');
```
I understand `{nothing}` should throw a warning because there is no `'nothing'` key in the context parameter

**Comment by Seldaek** [comment]
That is true.. although IMO it's not really practical for performance reasons to match placeholders in messages vs replace what's in the context keys if anything is there. So I am not sure what to do about it. It's kind of a minor issue but strictly speaking you're right.

**Comment by Maks3w** [comment]
I have created a project for help to message producers to verify their messages follows PSR-3 syntax.

It's like the TestCase shipped in this project but focus on producers instead of log libraries (consumers)

Could you give it a review https://github.com/Maks3w/Psr3MessageAssertions?
